atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchManagerImpl.java

```diff
@@ -55,12 +57,20 @@ public LockWatchManagerImpl(LockWatchEventCache lockWatchEventCache,
     }
 
     CommitUpdate getCommitUpdate(long startTs) {
-        return lockWatchEventCache.getCommitUpdate(startTs);
+        try {
+            return lockWatchEventCache.getCommitUpdate(startTs);
+        } catch (LockWatchFailedException e) {
+            throw new TransactionLockWatchFailedException("Failed to get commit update", e);
+        }
```

Comment:
It's a bit awkward, but the only way to properly put these as `TransactionFailedRetriableExceptions` given the awkward dependency loop.

Reply:
Can you explain the dependency loop? This is weird and there should be a different solution.

Reply:
`TransactionFailedRetriableException`, and others like it, live in `com.palantir.atlasdb.transaction.api`, which depends on `lock-api` (which in turn depends on `lock-api-objects`). To use retriable exceptions in this piece would require `lock-api` to depend on `com.palantir.atlasdb.transaction.api`, which creates a loop.

Something needs to move to avoid this awkwardness.

Reply:
That's what I mean, you should probably move the event cache code somewhere else. It feels weird that it's in an *-api subproject, where it clearly is an implementation?

Reply:
`com.palantir.atlasdb.keyvalue.api.watch` is used for `LockWatchManagerImpl` so clearly the impl in `*-api` is not adhered to.

Still, absolutely not averse to moving it - I will try and see where makes sense (without break decencies on the things that use it).

Reply:
Moved to the same place as `LockWatchManagerImpl` - I think this is a better place.